stdlib/LinearAlgebra/src/matmul.jl

```diff
@@ -1007,3 +1007,58 @@ function matmul3x3!(C::AbstractMatrix, tA, tB, A::AbstractMatrix, B::AbstractMat
     end # inbounds
     C
 end
+
+# Three-argument *
+*(A::AbstractMatrix, B::AbstractMatrix, x::AbstractVector) = _mat_mat_vec(A,B,x)
+
+*(α::Number, A::AbstractMatrix, x::AbstractVector) = _scalar_mat_vec(α,A,x)
+*(α::Number, A::AbstractMatrix, B::AbstractMatrix) = _scalar_mat_mat(α,A,B)
+*(A::AbstractMatrix, x::AbstractVector, α::Number) = _scalar_mat_vec(α,A,x)
+*(A::AbstractMatrix, B::AbstractMatrix, α::Number) = _scalar_mat_mat(α,A,B)
+
+*(α::Number, A::AbstractArray, β::Number) = (α*β) * A
+*(A::AbstractArray, α::Number, β::Number) = A * (α*β)
+
+*(α::Number, u::AbstractVector, tv::AdjOrTransAbsVec) = broadcast(*, α, u, tv)
+*(u::AbstractVector, tv::AdjOrTransAbsVec, α::Number) = broadcast(*, u, tv, α)
+*(u::AbstractVector, tv::AdjOrTransAbsVec, A::AbstractMatrix) = u * (tv*A)
+
+*(A::AbstractMatrix, B::AbstractMatrix, C::AbstractMatrix) = _tri_matmul(A,B,C)
+*(tv::AdjOrTransAbsVec, B::AbstractMatrix, C::AbstractMatrix) = (tv*B) * C
```

Comment:
Nitpicking, but this should be correctly covered by the 3-matrix-method one line above, right?

Reply:
Yes. I just thought it a little neater to do it by dispatch when you can, and `size` when you can't.